Source/WebCore/Modules/applicationmanifest/ApplicationManifestParser.cpp

```diff
@@ -173,6 +174,32 @@ URL ApplicationManifestParser::parseStartURL(const JSON::Object& manifest, const
     return startURL;
 }
 
+ApplicationManifest::Direction ApplicationManifestParser::parseDir(const JSON::Object& manifest)
+{
+    auto value = manifest.getValue("dir"_s);
+    if (!value)
+        return ApplicationManifest::Direction::Auto;
+
+    auto stringValue = value->asString();
+    if (!stringValue) {
+        logManifestPropertyNotAString("dir"_s);
+        return ApplicationManifest::Direction::Auto;
+    }
+
+    static constexpr std::pair<ComparableLettersLiteral, ApplicationManifest::Direction> directionMappings[] = {
+        { "auto", ApplicationManifest::Direction::Auto },
+        { "ltr", ApplicationManifest::Direction::Ltr },
+        { "rtl", ApplicationManifest::Direction::Rtl },
+    };
+    static constexpr SortedArrayMap directions { directionMappings };
+
+    if (auto* dirValue = directions.tryGet(StringView(stringValue).trim(isUnicodeCompatibleASCIIWhitespace<UChar>)))
```

Comment:
Using `isUnicodeCompatibleASCIIWhitespace` is almost certainly a bug. What kind of whitespace does the specification call for?

Reply:
I have adopted this line of code to ensure consistency with the parsing of [display modes](https://github.com/WebKit/WebKit/blob/044553fa65f400aee0001dcf2177cf78b49a53f5/Source/WebCore/Modules/applicationmanifest/ApplicationManifestParser.cpp#L223) and [orientations](https://github.com/WebKit/WebKit/blob/044553fa65f400aee0001dcf2177cf78b49a53f5/Source/WebCore/Modules/applicationmanifest/ApplicationManifestParser.cpp#L255). The trimming behavior is also covered by tests ([example](https://github.com/WebKit/WebKit/blob/044553fa65f400aee0001dcf2177cf78b49a53f5/Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp#L496)). The specification does not provide guidance on handling whitespace.

Reply:
Yeah, the spec doesn't trim anymore... we did some archeology... the spec used to trim back in 2017, but then it got dropped when we converted things to WebIDL. 

Discussed with @annevk.... we probably want to do this here:
 
```suggestion
    if (auto* dirValue = directions.tryGet(StringView(stringValue).trim(isASCIIWhitespace<UChar>)))
```

Reply:
Filed bug for the others... https://bugs.webkit.org/show_bug.cgi?id=278377